Permissions/Permissions/Private/Main.cpp

```diff
@@ -86,23 +91,43 @@ namespace Permissions
 
 		return AddPlayerToGroup(eos_id, group);
 	}
-
+	void HandleMessage(APlayerController* player_controller, auto result, auto message) {
```

Comment:
Slightly cleaned up and you should pass the existing `shooter_controller` from each command that is calling HandleMessage since it has already been cast

```
void HandleMessage(AShooterPlayerController* shooter_controller, auto result, auto message)
{
	if (HideAllPlayerSuccessMessages)
		return;
	
	if (Result.has_value())
	{
		if (!SendMessagesAsNotification)
			AsaApi::GetApiUtils().SendServerMessage(shooter_controller, FColorList::Red, result.value().c_str());
		else
			AsaApi::GetApiUtils().SendNotification(shooter_controller, FColorList::Red, TextSize, DisplayTime, nullptr, result.value().c_str());
	}
	else
	{
		if (!SendMessagesAsNotification)
			AsaApi::GetApiUtils().SendServerMessage(shooter_controller, FColorList::Green, message);
		else
			AsaApi::GetApiUtils().SendNotification(shooter_controller, FColorList::Green, TextSize, DisplayTime, nullptr, message);
	}
}
```

Reply:
Another option is to pass the `APlayerController` as you have it and remove all the casts in the commands since they no longer use the casted `shooter_conroller` locally anymore and then the `HandleMessage` is the only place the cast needs to exist since it is only needed for messaging.

Reply:
> Slightly cleaned up and you should pass the existing `shooter_controller` from each command that is calling HandleMessage since it has already been cast
> 
> ```
> void HandleMessage(AShooterPlayerController* shooter_controller, auto result, auto message)
> {
> 	if (HideAllPlayerSuccessMessages)
> 		return;
> 	
> 	if (Result.has_value())
> 	{
> 		if (!SendMessagesAsNotification)
> 			AsaApi::GetApiUtils().SendServerMessage(shooter_controller, FColorList::Red, result.value().c_str());
> 		else
> 			AsaApi::GetApiUtils().SendNotification(shooter_controller, FColorList::Red, TextSize, DisplayTime, nullptr, result.value().c_str());
> 	}
> 	else
> 	{
> 		if (!SendMessagesAsNotification)
> 			AsaApi::GetApiUtils().SendServerMessage(shooter_controller, FColorList::Green, message);
> 		else
> 			AsaApi::GetApiUtils().SendNotification(shooter_controller, FColorList::Green, TextSize, DisplayTime, nullptr, message);
> 	}
> }
> ```

The way it was setup was to still allow for error messages to be logged. Doing it this way would suppress all messages.